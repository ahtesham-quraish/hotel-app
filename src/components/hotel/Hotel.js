import React from 'react';
import pure from 'recompose/pure';
const Hotel = (props) => {
    const { hotel } = props;
    return (
      <li>
        <span className="info">
          <span>Name:</span> {hotel.name}{' '}
        </span>
        <span className="info">
          <span>Price:</span> {`${hotel.price} AED`}{' '}
        </span>
        <span className="info">
          <span>City:</span> {hotel.city}{' '}
        </span>
      </li>
    );
}
export default pure(Hotel);
