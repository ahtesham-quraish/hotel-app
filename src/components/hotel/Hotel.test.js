import React from 'react';
import ReactDOM from 'react-dom';
import Hotel from './Hotel';
import Enzyme, { shallow, render, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import fetchHotelList from '../../../__mocks__/fetchHotels';
Enzyme.configure({ adapter: new Adapter() });

it('renders without crashing', async () => {
  const hotelList = await fetchHotelList();
  const wrapper = shallow(<Hotel hotel={hotelList.hotels[0]} />);
  expect(wrapper).toMatchSnapshot();
});
test('render a document name', async () => {
  const hotelList = await fetchHotelList();
  const wrapper = mount(<Hotel hotel={hotelList.hotels[0]} />);
  expect(wrapper.prop('hotel').name).toEqual(hotelList.hotels[0].name);
});
