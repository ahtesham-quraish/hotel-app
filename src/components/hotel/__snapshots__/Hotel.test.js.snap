// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <pure(Hotel)
    hotel={
        Object {
            "availability": Array [
              Object {
                "from": "10-10-2020",
                "to": "15-10-2020",
              },
              Object {
                "from": "25-10-2020",
                "to": "15-11-2020",
              },
              Object {
                "from": "10-12-2020",
                "to": "15-12-2020",
              },
            ],
            "city": "dubai",
            "name": "Media One Hotel",
            "price": 102.2,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "hotel": Object {
        "availability": Array [
          Object {
            "from": "10-10-2020",
            "to": "15-10-2020",
          },
          Object {
            "from": "25-10-2020",
            "to": "15-11-2020",
          },
          Object {
            "from": "10-12-2020",
            "to": "15-12-2020",
          },
        ],
        "city": "dubai",
        "name": "Media One Hotel",
        "price": 102.2,
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "hotel": Object {
          "availability": Array [
            Object {
              "from": "10-10-2020",
              "to": "15-10-2020",
            },
            Object {
              "from": "25-10-2020",
              "to": "15-11-2020",
            },
            Object {
              "from": "10-12-2020",
              "to": "15-12-2020",
            },
          ],
          "city": "dubai",
          "name": "Media One Hotel",
          "price": 102.2,
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
